### ISSUE TITLE
Rubocop generates invalid code

### ISSUE THREAD

**Issue by adfoster-r7** [rubocop-hq/rubocop#7776]
When formatting the following code, rubocop will generate invalid ruby:

```ruby
class Bar
  def initialize(*args)
  end
end

class Foo < Bar
  def initialize
    super(foo(bar,
      'ExpiresAt' => {'Hours' => 60 * 60 * 24} # 24h
    ))
  end
end
```

## Expected behavior

Not to generate invalid ruby code

## Actual behavior

Invalid ruby code is created due to the comment handling of `# 24 h`

```ruby
# frozen_string_literal: true
class Bar
  def initialize(*args); end
end

class Foo < Bar
  def initialize
    super(foo(bar,
              'ExpiresAt' => { 'Hours' => 60 * 60 * 24 }) # 24h)
  end
end
```

## Steps to reproduce the problem

1. Create new ruby file
2. add original code snippet
3. bundle exec rubocop -a new_file.rb
4. Run again, verify invalid ruby code has been generated

## RuboCop version

```
$ [bundle exec] rubocop -V
0.80.0 (using Parser 2.7.0.2, running on ruby 2.6.5 x86_64-darwin18)
```

**Comment by bmorrall** [comment]
Had a quick look at this, running autocorrect with only the `Layout/MultilineMethodCallBraceLayout` Cop produces this error.

**Comment by MattBeaty** [comment]
I'd like to help out on this project. Mind if I take a crack at this issue?
